### PR TITLE
Fix mark release ready

### PR DESCRIPTION
**Pull request by Felixoid**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
The GH actions have a very ugly way of checking if all 1st level dependencies succeeded. The approach is tested in https://github.com/Felixoid/actions-experiments/commit/65f6be814e1eb143149fe7910742c9b518872e97

Besides that, unlock the way to mark any commit as ready for release locally.

### CODE REVIEW

**Comment by robot-ch-test-poll** [comment]
*This is an automated comment for commit a5e3b7982ac310591c815b37541926ce26bdb133 with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/59994/a5e3b7982ac310591c815b37541926ce26bdb133/ci_running.html) to open a full report in a separate page

<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/59994/a5e3b7982ac310591c815b37541926ce26bdb133/ci_running.html">⏳ pending</a></td></tr>
<tbody>
</table>

**Comment by Felixoid** [comment]
The output in the job logs:

```
needs data
{
  "CreateArtifact": {
    "result": "success",
    "outputs": {}
  },
  "WillSkip": {
    "result": "skipped",
    "outputs": {}
  },
  "WillFail": {
    "result": "failure",
    "outputs": {}
  }
}
[
  "success",
  "skipped",
  "failure"
]
no failures false
no skips false
no both false
```
https://github.com/Felixoid/actions-experiments/actions/runs/7905115935/job/21576986609

File: .github/workflows/master.yml [hunk 1]
```diff
@@ -327,6 +327,7 @@ jobs:
       run_command: |
         python3 build_report_check.py "$CHECK_NAME"
   MarkReleaseReady:
+    if: ${{ ! (contains(needs.*.result, 'skipped') || contains(needs.*.result, 'failure')) }}
```
Comment (maxknv) [inline .github/workflows/master.yml:330]: ${{ !failure() && !cancelled() }} should work

Comment (Felixoid) [reply .github/workflows/master.yml:330]: it's either broken in https://github.com/ClickHouse/ClickHouse/actions/runs/7896315258/job/21575897913 or in https://github.com/ClickHouse/ClickHouse/actions/runs/7904523295/job/21577047648

We should explicitly check that every needed job succeeded. Not the tree of all dependencies

**Comment by Felixoid** [comment]
Discussed offline, pushing yesterday version

**Pull request merged**